### ISSUE TITLE
Display scale compensation

### ISSUE THREAD

**Issue by Kamik423** [JohnCoates/Aerial#821]
With two displays not having the same scaling the features are different sizes. I have my 5k iMac monitor set to the smallest possible scaling (smallest text – most content) and an external 1080p monitor set to native scale, which makes text and the screensaver content appear much larger on it.

I assume this is due to the screensaver using pixels/points as reference sizes and not physical size. Maybe an option can be added to correct for that. So if a feature spanning both monitors moves one point/pixel on each monitor the physical distance will be much bigger on the external one.

As you can see in the attached ~~video~~ gif the monitor on the left is scrolling a lot faster than the one on the right.

![ezgif-1-8988ec1dc14e](https://user-images.githubusercontent.com/16278233/60094249-d39af500-974b-11e9-97e6-3363a4f8baf4.gif)

**Comment by glouel** [comment]
Hi @Kamik423 

That screenshot you sent earlier is from the same system, right ? 

<img width="780" alt="Screen Shot 2019-06-24 at 20 15 13" src="https://user-images.githubusercontent.com/16278233/60041922-c71b8b80-96bc-11e9-8366-68a06290cba1.png">

To give you an understanding, your 2 screens are seen by macOS as one large virtual screen in points, and the default scaling of each screen (pixels to points) is calculated so it looks "right" even if you have one retina screen, and one non retina screen. 

So yes, I'm using that same ratio you see in either Aerial's config panel or System Preferences Display options, and show a portion of the screen based on those sizes as seen. That's why Aerial think your left screen is much smaller than it is (so does macOS). Because that's the only info we accurately have that takes into account position, etc, that's what Aerial uses.

**Comment by Kamik423** [comment]
What I am asking for is to add in a field so we can supply the missing physical information by measuring the height of the display or maybe even add a small database of commonly used screens so that can be corrected.

**Comment by glouel** [comment]
I understand what you want but this is pretty complex to do correctly, as the only accurate information we have is how macOS thinks your screen looks like relative to each others. 

I can try to have a look at some point but when you start considering more than 2 monitors, their relative positions, etc, that's a lot of work to get exactly right in all cases, and not something I can add quickly.

**Comment by Kamik423** [comment]
I understand. Still hoping for it at some point in the future! 

Maybe a visual (approximate) rescale tool, similar to the monitor positioning system dialog, but with resizing and free movement would be the simplest and most user-friendly way to implement that scalably.

**Comment by glouel** [comment]
Yes you'd need to decouple it all and that kind of UI would work

If you set your iMac scaling to default, their relative size will look the same or close enough, you can check how it works for another user here : https://github.com/JohnCoates/Aerial/issues/194#issuecomment-491516236

That info is what I get from macOS. I can separately get for each screen an estimation of their size in centimeters but that doesn't take into account positionning so that's why you'd need to have something to set that too.

I'm not saying never, but right now I'm more in firefighting more than anything because of Catalina. I marked this as an enhancement at a future point to have a look, but I don't want to get your hopes up that it will come in next build ;)

**Comment by Kamik423** [comment]
No pressure – I am just glad your are making this at all. Keep up the great work and extinguish those fires!

**Comment by duecedriver** [comment]
once again.. you have to do what Adobe does for photoshop and its other professional apps

you have to be able to input the precise dpi of the monitors so that for example a 100x100 px block drawn across 2 different monitors of 2 different physical sizes and DPI are of the same physical height
